src/main/java/com/revlearn/team1/config/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,75 @@
+package com.revlearn.team1.config;
+
+import com.revlearn.team1.exceptions.UserNotFoundException;
+import com.revlearn.team1.model.User;
+import com.revlearn.team1.service.user.UserService;
+import com.revlearn.team1.util.JwtUtil;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private final UserDetailsService userDetailsService;
+    private final JwtUtil jwtUtil;
+    private final UserService userService;
+
+    @Override
+    protected boolean shouldNotFilter(@NonNull HttpServletRequest request) throws ServletException {
+        final List<String> skipFilterUrls = Arrays.asList("/auth/**");
+        return skipFilterUrls.stream().anyMatch(url -> new AntPathRequestMatcher(url).matches(request));
+    }
+
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request, @NonNull HttpServletResponse response, @NonNull FilterChain filterChain) throws ServletException, IOException {
+        final String authHeader = request.getHeader("Authorization");
+        final String jwt;
+        final Long userId;
+
+        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
+            filterChain.doFilter(request, response);
+            System.out.println("No auth header");
+            return;
+        }
+
+        jwt = authHeader.substring(7);
+
+        userId = jwtUtil.extractUserId(jwt);
```

Comment:
can we add a null check here just in case of malformed token

Reply:
I think any issue with a malformed token will be handled by the Jwts package/dependency.  It is called inside our JwtUtil class:
    public Claims decodeJWT(String token) {
        try {
            return Jwts.parserBuilder()
                    .setSigningKey(Keys.hmacShaKeyFor(Base64.getDecoder().decode(secretKey)))
                    .build()
                    .parseClaimsJws(token)
                    .getBody();
        } catch (ExpiredJwtException | MalformedJwtException | UnsupportedJwtException | SignatureException |
                 WeakKeyException | IllegalArgumentException e) {
            throw new RuntimeException("Invalid JWT token. " + e.getMessage());
        }
    }